### PR TITLE
extract `uniqBy` as generic method and reuse internally

### PR DESCRIPTION
**Pull request by bekzod**
@mmun is this worth doing ? 😉

### CODE REVIEW

File: packages/ember-runtime/lib/utils.js [hunk 1]
```diff
@@ -168,3 +169,21 @@ export function typeOf(item) {
 
   return ret;
 }
+
+export function uniqBy(array, key) {
+  assert(`first argument passed to \`uniqBy\` should be array`, isArray(array));
+
+  let ret = A();
+  let seen = new Set();
+  let getter = arguments.length === 1 ? item => item : get;
```
Comment (mmun) [inline packages/ember-runtime/lib/utils.js]: can you do something like
```js
const identityFunction = item => item;
export function uniqBy(array, key = identityFunction) {
```

Comment (bekzod) [reply packages/ember-runtime/lib/utils.js]: alright

**Review by mmun — APPROVED** [review]
This seems fine to do.

**Comment by bekzod** [comment]
green

**Comment by rwjblue** [comment]
Awesome, thank you!

**Pull request merged**